src/audio/module_adapter/module/memory-common.c

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+
+/*
+ * \file generic.c
+ * \brief Generic Codec API
+ * \author Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ *
+ */
+
+#include <rtos/symbol.h>
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/data_blob.h>
+#include <sof/lib/fast-get.h>
+
+LOG_MODULE_DECLARE(module_adapter, CONFIG_SOF_LOG_LEVEL);
+
+int module_init(struct processing_module *mod)
```

Comment:
This function does not appear to be related to memory. Can you explain why it was moved?

Reply:
This one doesn't do any memory allocations directly, they are done via other calls so its quite generic.